Add versioning to MTEB benchmarks

KennethEnevoldsen opened the pull request:
- Following suggestion made in #2001 I added version to MTEB benchmarks
- changed the name of the MTEB(Chinese) to MTEB(cmn, v1). Though we could go for MTEB(Chinese, v1) assuming it is a group (also covering chinese other than mandarin)
- change the name of FaMTEB(fas, beta) to MTEB(fas)
- did a minor formatting of imports due to a circular import error
- moved the Benchmark object out of the file with the benchmarks
  - this is still >1000 lines so we could split it up to "external_benchmarks", "monolingual_benchmarks", "domain_specific_benchmarks", "multilingual_benchmarks"

Fixes #2001


<!-- If you are submitting a dataset or a model for the model registry please use the corresponding checklists below otherwise feel free to remove them. -->

<!-- add additional description, question etc. related to the new dataset -->


### Code Quality
<!-- Please do not delete this -->
- [x] **Code Formatted**: Format the code using `make lint` to maintain consistent style.

### Documentation
<!-- Please do not delete this -->
- [ ] ~~**Updated Documentation**: Add or update documentation to reflect the changes introduced in this PR.~~

### Testing
<!-- Please do not delete this -->
- [ ] **New Tests Added**: Write tests to cover new functionality. Validate with `make test-with-coverage`.
- [ ] **Tests Passed**: Run tests locally using `make test` or `make test-with-coverage` to ensure no existing functionality is broken.

Muennighoff commented on mteb/benchmarks/benchmarks.py:
```diff
@@ -1219,7 +1159,7 @@ def load_results(
 
 C_MTEB = Benchmark(
-    name="MTEB(Chinese)",
+    name="MTEB(cmn, v1)",
```
Do we know all these tasks are Mandarin? If so then should we also rename the folders to `cmn`? 
[zho](https://github.com/embeddings-benchmark/mteb/tree/main/mteb/tasks/Retrieval/zho)

Samoed replied:
All tasks in metadata have `cmn` as thier language https://github.com/embeddings-benchmark/mteb/blob/1588b9ac31b7a84d1e34ef0325c0c10b0b0e8db5/mteb/tasks/Retrieval/zho/CMTEBRetrieval.py#L96C22-L96C25

Muennighoff replied:
I guess we can also keep the folder as zho and then use it for tasks with other dialects too

Muennighoff reviewed (commented):
Nice! To avoid breaking backward compat would it make sense to temporarily allow the old names by just mapping to the new benchmark carrying that name and raising a warning that the old name will be deprecated soon?

x-tabdeveloping reviewed (commented):
I agree with Niklas, Also make sure this doesn't break the leaderboard (I think it probably will since the multilingual benchmark is referred to by name)

KennethEnevoldsen commented:
I have added changed for backward compatibility. I have also updated the leaderboard code to refer to the `MTEB_multlingual.name`. This should prevent inconsistencies.

Samoed commented on mteb/benchmarks/get_benchmark.py:
```diff
@@ -1,20 +1,68 @@
+    if benchmark_name in PREVIOUS_BENCHMARK_NAMES:
+        logger.warning(
+            f"Using the previous benchmark name '{benchmark_name}' is deprecated. Please use '{PREVIOUS_BENCHMARK_NAMES[benchmark_name]}' instead."
+        )
```
I think now, exception would be raised.
```suggestion
        )
        benchmark_name = PREVIOUS_BENCHMARK_NAMES[benchmark_name]
```
